ff/smirnoff_converter.py

```diff
@@ -85,95 +88,109 @@ def parse_quantity(number_string):
+                patt = s.attributes['smirks'].value
+                b0 = parse_quantity(s.attributes['length'].value)
+                kb = parse_quantity(s.attributes['k'].value)
+                params.append([patt, kb, b0])
+            bonds = {
+                "patterns": params,
+            }
+            forcefield["HarmonicBond"] = bonds
+
+        elif tag == ANGLE_TAG:
+            params = []
+            for s in itemlist:
+                patt = s.attributes['smirks'].value
+                a0 = parse_quantity(s.attributes['angle'].value)
+                ka = parse_quantity(s.attributes['k'].value)
+                params.append([patt, ka, a0])
+            angles = {
+                "patterns": params,
+            }
+            forcefield["HarmonicAngle"] = angles
+        elif tag == PROPER_TAG:
+            params = []
+            for s in itemlist:
+                patt = s.attributes['smirks'].value
+                counter = 1
+                components = []
+                while True:
+                    try:
+                        k = parse_quantity(s.attributes['k'+str(counter)].value)
+                        phase = parse_quantity(s.attributes['phase'+str(counter)].value)
+                        period = float(s.attributes['periodicity'+str(counter)].value)
+                        idivf = float(s.attributes['idivf'+str(counter)].value)
+                        k = k/idivf
+                        components.append([k, phase, period])
+                        counter += 1
+                    except KeyError:
+                        break
+                params.append([patt, components])
+            torsions = {
+                "patterns": params,
+            }
+            forcefield["ProperTorsion"] = torsions
+        elif tag == IMPROPER_TAG:
+            params = []
+            for s in itemlist:
+                patt = s.attributes['smirks'].value
+                impdivf = 3
+                k = parse_quantity(s.attributes['k1'].value)/impdivf
+                phase = parse_quantity(s.attributes['phase1'].value)
+                period = float(s.attributes['periodicity1'].value)
+                params.append([patt, k, phase, period])
+            impropers = {
+                "patterns": params
+            }
+            forcefield["ImproperTorsion"] = impropers
+        elif tag == VDW_TAG:
+            params = []
+            for s in itemlist:
+                patt = s.attributes['smirks'].value
+                epsilon = parse_quantity(s.attributes['epsilon'].value)
+                if "rmin_half" in s.attributes:
+                    rmin_half = parse_quantity(s.attributes['rmin_half'].value)
+                    sigma = 2. * rmin_half / (2.**(1. / 6.))
+                else:
+                    sigma = parse_quantity(s.attributes['sigma'].value)
+                # Take sqrt of epislon to avoid singularity in backprop
+                params.append([patt, sigma, np.sqrt(epsilon)])
```

Comment:
Hard to tell what the actual diffs are due to indentation, but I take it this is the only one that changed?

Reply:
The output keys for the FF are also different. Previously they resulted in a dict that the keys had to be renamed. I don't know if that was necessary historically or if there is a secondary stage of conversion I was meant to run. Otherwise this is the only substantive change to the script.